### PR TITLE
Checkout: Thank you: Show a different page for domain mappings

### LINKED ISSUE #9818: Checkout: incorrect notice shown after mapping domains already using our name servers
After adding a domain mapping, we seem to always show this notice, regardless of the actual state of the domain:

ranimac3 screenshot 2016-12-05 at 12 50 41

This notice is only relevant when the domain is not already using our name servers. If the domain is already correctly set up before mapping it, the notice is incorrect and confusing. This may happen if users read our [docs for domain mapping](https://en.support.wordpress.com/map-existing-domain/) and make the necessary settings in advance, if their domain was set up automatically to use WordPress.com name servers by get.blog, if the domain was already mapped to a WordPress.com site in the past and the name servers were never changed, or for other reasons.

We should only show this notice when the domain is not already correctly configured to work with WordPress.com. 

#### Steps to reproduce
1. Have a domain that is set to use WordPress.com name servers and that is not already mapped to WordPress.com.
2. Add domain mapping for this domain.
3. After paying, you will see a notice like the one above.

### PR DESCRIPTION
**Pull request by scruffian**
This pull request fixes #9818 by showing a generic success page for a domain mappings that are already mapped to a WordPress.com site:

screen shot 2016-12-09 at 21 28 45
  
#### Testing instructions
  
1. Run `git checkout **`** and start your server
1. Apply this patch: D3643-code
2. Purchase a domain mapping for a domain which has name servers pointing to ns1.wordpress.com
3. Check that you see the success page as above
2. Purchase a domain mapping for a domain which DOES NOT HAVE name servers pointing to ns1.wordpress.com
3. Check that you see the regular domain mapping success page (see #9818)  
  
#### Reviews
  
- [ ] Code
- [ ] Product
   
@Automattic/sdev-feed

### CODE REVIEW

File: client/lib/products-values/index.js [hunk 1]
```diff
@@ -217,6 +217,13 @@ function isDomainMapping( product ) {
 	return product.product_slug === 'domain_map';
 }
 
+function isDomainMappingNotMappedToWordPressDotCom( product ) {
```
Comment (yurynix) [inline client/lib/products-values/index.js]: I'm a bit confused, the name of the function suggests the domain is *not* mapped to WordPress.com, but
what is checked that it is actually mapped:

```js
return product.product_slug === 'domain_map' && product.mappedDomainHasWpcomNameServers
```

What am I missing?

Comment (scruffian) [inline client/lib/products-values/index.js]: Nothing, i just named the function very badly!

**Comment by fabianapsimoes** [comment]
Seems I'm seeing the same version of the Thank You page (the one being introduced in this PR) for both cases. I do see the "DNS configuration required" banner in the Site Picker, but the Thank You page was the same.

**Comment by gziolo** [comment]
@fabianapsimoes @scruffian: It's ready for another review.

**Pull request closed without merging**